### PR TITLE
[KOGITO-1023] consolidate kogito-bom and kogito-runtimes repos

### PR DESCRIPTION
**Pull request by mariofusco**

### CODE REVIEW

**Comment by mariofusco** [comment]
@mswiderski @cristianonicolai I believe this PR is failing because the old bom project is still built before of this. Could you please give a look and let me know if I should do something about this?

**Comment by mswiderski** [comment]
@mariofusco try removing this stage https://github.com/kiegroup/kogito-runtimes/blob/master/Jenkinsfile#L25-L34

File: boms/Jenkinsfile.deploy [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+            script {
+                mailer.sendEmailFailure()
+            }
+        }
+        always {
+            junit '**/target/surefire-reports/**/*.xml'
+            cleanWs()
+        }
+    }
+}
```
Comment (mareknovotny) [inline boms/Jenkinsfile.deploy]: this file doesn't need to be merged as always we will have only Jenkinsfile in root of repository

File: boms/scripts/mvn-all.sh [hunk 2]
```diff
@@ -0,0 +1,127 @@
+#!/bin/bash
```
Comment (mareknovotny) [inline boms/scripts/mvn-all.sh]: we should probably move scripts folder to the root of repository not to boms folder

**Comment by mareknovotny** [comment]
I think  we should have boms on the first level of repository and merge the bom-parent with kogito-runtimes parent.

**Comment by mariofusco** [comment]
jenkins test

**Comment by mariofusco** [comment]
@mareknovotny I moved the scripts one folder up as you suggested. As for merging the kogito-bom-parent into kogito-runtimes-parent, I believe that we will need a kogito-bom-parent regardless as root module for kogito-bom and kogito-deps-bom and since I cannot remove it I guess it will be cleaner to keep bom related stuff in that pom. What do you think?

File: scripts/repository-list.txt [hunk 3]
```diff
@@ -0,0 +1,4 @@
+kogito-bom
```
Comment (mareknovotny) [inline scripts/repository-list.txt]: this should be removed too now

**Comment by mareknovotny** [comment]
I don't follow why the bom-parent is good to have in boms folder. 
Now root of kogito-runtimes has defined version properties so the kogito-bom-parent is useless imho. If we move individual boms into root it will be simpler.

File: boms/Jenkinsfile [hunk 4]
```diff
@@ -0,0 +1,83 @@
+@Library('jenkins-pipeline-shared-libraries')_
```
Comment (mswiderski) [inline boms/Jenkinsfile]: this is not needed here, jenkinsfile is only needed in top level repo

File: boms/Jenkinsfile.deploy [hunk 5]
```diff
@@ -0,0 +1,75 @@
+@Library('jenkins-pipeline-shared-libraries')_
```
Comment (mswiderski) [inline boms/Jenkinsfile.deploy]: same here, not needed file

File: boms/README.md [hunk 6]
```diff
@@ -0,0 +1 @@
+# kogito-bom
```
Comment (mswiderski) [inline boms/README.md]: since this is empty file maybe we don't need to at all, no longer it's own repo

**Comment by mariofusco** [comment]
@mareknovotny better?

**Comment by mareknovotny** [comment]
it is better @mariofusco almost ok, but one fact is that you merged kogito-bom and kogito-deps-bom which could make some trouble if user will import that and have a need to use some other 3rd party dependencies in their projects.

**Comment by MarianMacik** [comment]
> so if i've understood correctly, now all submodules, except for bom (of course) are _importing_ bom -- instead of it being their parent. I am not an expert I don't think there are downsides (?) Anyway, we can always improve this later.

@evacchi 
One downside of it is that now you cannot override a dependency version (if needed) with just a simple Maven property. You have to declare a full `dependencyManagement` tag.

**Comment by mareknovotny** [comment]
we can merge it as it is to not block this consolidation

**Comment by cristianonicolai** [comment]
@mareknovotny should https://github.com/kiegroup/kogito-bom be archived?

**Comment by Ginxo** [comment]
@cristianonicolai @mareknovotny @mariofusco the rest of the projects still inheriting from `kogito-bom-parent` and PRs like https://github.com/kiegroup/kogito-cloud/pull/89 or https://github.com/kiegroup/kogito-examples/pull/130/files don't seem to solve it. Am I missing something?

**Comment by mareknovotny** [comment]
@Ginxo I think the follow up PRs are not merged, but they should be quickly finished to have all repos consistent

**Comment by mareknovotny** [comment]
@cristianonicolai I archived kogito-bom now

**Pull request merged**